building model_explorer example: experimental_onnxruntime_cxx_api.h; errors

**Describe the bug**

Greetings. How should I compile the model-explorer example in samples/c_cxx? (The goal behind it is to build further programs using the c++ API for inference.)

After building the onnxruntime itself (on Linux), I went to the model_builder folder and tried to build via cmake. (model_builder appears to be the only non-Windows-only c++ sample).

The compiler complained about needing "experimental_onnxruntime_cxx_api.h", so I added the include directory via CXXFLAGS and rebuilt. 

Then, the compiler complained that "-std=c++11" was needed, so I added that via CXXFLAGS, too.

Then, the following compiler errors occur:

In file included from /home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:39:0:
/home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h:22:14: error: expected ‘{’ before ‘::’ token
 namespace Ort::Experimental {
              ^
/home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h:22:16: error: ‘Experimental’ in namespace ‘::’ does not name a type
 namespace Ort::Experimental {
                ^
In file included from /home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h:67:0,
                 from /home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:39:
/home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_inline.h:8:14: error: expected ‘{’ before ‘::’ token
 namespace Ort::Experimental {
              ^
/home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_inline.h:8:16: error: ‘Experimental’ in namespace ‘::’ does not name a type
 namespace Ort::Experimental {
                ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp: In function ‘int Ort::Ort::main(int, char**)’:
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:66:3: error: ‘Env’ is not a member of ‘Ort::Ort’
   Ort::Env env(ORT_LOGGING_LEVEL_WARNING, "batch-model-explorer");
   ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:66:3: note: suggested alternative:
In file included from /home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h:20:0,
                 from /home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:39:
/home/user/onnxruntime/include/onnxruntime/core/session/onnxruntime_cxx_api.h:250:8: note:   ‘Ort::Env’
 struct Env : Base<OrtEnv> {
        ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:67:3: error: ‘SessionOptions’ is not a member of ‘Ort::Ort’
   Ort::SessionOptions session_options;
   ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:67:3: note: suggested alternative:
In file included from /home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h:20:0,
                 from /home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:39:
/home/user/onnxruntime/include/onnxruntime/core/session/onnxruntime_cxx_api.h:293:8: note:   ‘Ort::SessionOptions’
 struct SessionOptions : Base<OrtSessionOptions> {
        ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:68:8: error: ‘Ort::Ort::Experimental’ has not been declared
   Ort::Experimental::Session session = Ort::Experimental::Session(env, model_file, session_options);
        ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:71:22: error: ‘session’ was not declared in this scope
   auto input_names = session.GetInputNames();
                      ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:103:17: error: ‘Value’ is not a member of ‘Ort::Ort’
     std::vector<Ort::Value> batch_input_tensors;
                 ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:103:17: note: suggested alternative:
In file included from /home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h:20:0,
                 from /home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:39:
/home/user/onnxruntime/include/onnxruntime/core/session/onnxruntime_cxx_api.h:414:8: note:   ‘Ort::Value’
 struct Value : Base<OrtValue> {
        ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:103:17: error: ‘Value’ is not a member of ‘Ort::Ort’
     std::vector<Ort::Value> batch_input_tensors;
                 ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:103:17: note: suggested alternative:
In file included from /home/user/onnxruntime/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h:20:0,
                 from /home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:39:
/home/user/onnxruntime/include/onnxruntime/core/session/onnxruntime_cxx_api.h:414:8: note:   ‘Ort::Value’
 struct Value : Base<OrtValue> {
        ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:103:27: error: template argument 1 is invalid
     std::vector<Ort::Value> batch_input_tensors;
                           ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:103:27: error: template argument 2 is invalid
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:104:25: error: request for member ‘push_back’ in ‘batch_input_tensors’, which is of non-class type ‘int’
     batch_input_tensors.push_back(Ort::Experimental::Value::CreateTensor<float>(batch_input_tensor_values.data(), batch_input_tensor_values.size(), input_shape));
                         ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:104:40: error: ‘Ort::Ort::Experimental’ has not been declared
     batch_input_tensors.push_back(Ort::Experimental::Value::CreateTensor<float>(batch_input_tensor_values.data(), batch_input_tensor_values.size(), input_shape));
                                        ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:104:74: error: expected primary-expression before ‘float’
     batch_input_tensors.push_back(Ort::Experimental::Value::CreateTensor<float>(batch_input_tensor_values.data(), batch_input_tensor_values.size(), input_shape));
                                                                          ^
In file included from /home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:35:0:
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:107:33: error: invalid types ‘int[int]’ for array subscript
     assert(batch_input_tensors[0].IsTensor() &&
                                 ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:108:33: error: invalid types ‘int[int]’ for array subscript
            batch_input_tensors[0].GetTensorTypeAndShapeInfo().GetShape() == input_shape);
                                 ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:109:78: error: invalid types ‘int[int]’ for array subscript
     cout << "batch_input_tensor shape: " << print_shape(batch_input_tensors[0].GetTensorTypeAndShapeInfo().GetShape()) << endl;
                                                                              ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:120:34: error: expected unqualified-id before ‘&’ token
     } catch (const Ort::Exception& exception) {
                                  ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:120:34: error: expected ‘)’ before ‘&’ token
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:120:34: error: expected ‘{’ before ‘&’ token
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:120:45: error: expected primary-expression before ‘)’ token
     } catch (const Ort::Exception& exception) {
                                             ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp: At global scope:
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:126:1: error: expected ‘}’ at end of input
 }
 ^
/home/user/onnxruntime/samples/c_cxx/model-explorer/batch-model-explorer.cpp:126:1: error: expected ‘}’ at end of input
make[2]: *** [CMakeFiles/batch-model-explorer.dir/batch-model-explorer.cpp.o] Error 1
make[1]: *** [CMakeFiles/batch-model-explorer.dir/all] Error 2
make: *** [all] Error 2

Could anyone give tips for building this example?

**Urgency**
None.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Ubuntu 16.04.5 LTS
- ONNX Runtime installed from (source or binary): source from github
- GCC/Compiler version (if compiling from source): 5.4.0 20160609

--- a/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h
+++ b/include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h
@@ -19,7 +19,8 @@
 #pragma once
 #include "onnxruntime_cxx_api.h"
 
-namespace Ort::Experimental {
+namespace Ort {
+namespace Experimental {
 
 struct Session : Ort::Session {
   Session(Env& env, std::basic_string<ORTCHAR_T>& model_path, SessionOptions& options)
@@ -62,6 +63,7 @@ struct Value : Ort::Value {
   static Ort::Value CreateTensor(const std::vector<int64_t>& shape, ONNXTensorElementDataType type);
 };
 
-}  // namespace Ort::Experimental
+}
+}
 
 #include "experimental_onnxruntime_cxx_inline.h"

--- a/include/onnxruntime/core/session/experimental_onnxruntime_cxx_inline.h
+++ b/include/onnxruntime/core/session/experimental_onnxruntime_cxx_inline.h
@@ -5,7 +5,8 @@
 //
 // These are the inline implementations of the C++ header APIs. They are in this separate file as to not clutter
 // the main C++ file with implementation details.
-namespace Ort::Experimental {
+namespace Ort {
+namespace Experimental {
 
 inline std::vector<Ort::Value> Session::Run(const std::vector<std::string>& input_names, const std::vector<Ort::Value>& input_values,
                                             const std::vector<std::string>& output_names, const RunOptions& run_options) {
@@ -106,4 +107,5 @@ inline Ort::Value Value::CreateTensor(const std::vector<int64_t>& shape, ONNXTen
   return Ort::Value::CreateTensor(allocator, shape.data(), shape.size(), type);
 }
 
-}  // namespace Ort::Experimental
+}
+}

